### PR TITLE
update lambda part in client tool

### PR DESCRIPTION
**Pull request by nanmu08**
update lambda part in client tool, replace aws cli with paws lib, also update lambda part in faasr trigger file.

### CODE REVIEW

File: R/faasr_client_api_aws-lambda.R [hunk 1]
```diff
@@ -44,30 +44,41 @@
...
 
+# specify global variables
+# faasr_defined_image_lambda <- "145342739029.dkr.ecr.us-east-1.amazonaws.com/aws-lambda-tidyverse:latest"
+
 faasr_register_workflow_aws_lambda <- function(faasr, cred){
+
+  # get region from faasr
+  for (faas in names(faasr$ComputeServers)){
+    if(faasr$ComputeServers[[faas]]$FaaSType == "Lambda"){
+      aws_region <- faasr$ComputeServers[[faas]]$Region
```
Comment (spark0510) [inline R/faasr_client_api_aws-lambda.R:55]: I guess "aws_region" will be overwritten, if there are more than 2 Lambda Servers which are using different Region.

File: R/faasr_client_api_aws-lambda.R [hunk 2]
```diff
@@ -44,30 +44,41 @@
...
+      break
+    }
+  }
   
   
   # get aws lambda function list
-  lambda_function_info <- faasr_register_workflow_lambda_function_lists(faasr)
+  lambda_function_info <- faasr_register_workflow_lambda_function_lists(faasr, cred, aws_region)
   if (length(lambda_function_info)==0){
     return(FALSE)
```
Comment (spark0510) [inline R/faasr_client_api_aws-lambda.R]: Please make it retrun "" instead of FALSE

File: R/faasr_client_api_aws-lambda.R [hunk 3]
```diff
@@ -148,14 +159,24 @@ faasr_register_workflow_lambda_function_image <- function(faasr){
...
-  # get the AWS account ID
-  get_account_id_command <- "aws sts get-caller-identity --query Account --output text"
-  aws_account_id <- system(get_account_id_command, intern = TRUE)
+faasr_register_workflow_aws_ecr_build <- function(faasr, cred, aws_region){
+  sts_instance <- paws::sts(
+    config=list(
+      credentials=list(
+        creds=list(
+          access_key_id=cred$lambda_a_ACCESS_KEY,
+          secret_access_key=cred$lambda_a_SECRET_KEY,
```
Comment (spark0510) [inline R/faasr_client_api_aws-lambda.R]: "lambda_a_" is not defined.

Comment (spark0510) [reply R/faasr_client_api_aws-lambda.R]: Every functions for paws is using "lambda_a_" in their creds, so it needs to be modified.

File: R/faasr_client_api_aws-lambda.R [hunk 4]
```diff
@@ -473,13 +639,46 @@ faasr_set_workflow_timer_ld <- function(faasr, cred, target, cron, unset=FALSE){
...
+        )
+      ),
+      region=aws_region
+    )
+    )
+    # json file with credentials will be created and after invocation, it will be removed.
+    faasr_w_cred <- faasr_replace_values(faasr, cred)
+    faasr_json <- jsonlite::toJSON(faasr_w_cred, auto_unbox=TRUE)
+    rd_nb <- sample(100000, size=1)
+    writeLines(faasr_json, paste0("payload_ld_",rd_nb,".json"))
```
Comment (spark0510) [inline R/faasr_client_api_aws-lambda.R]: We don't need to create the "payload.json" file anymore.
line no.666, 668, and 669 will be unnecessary
line no.682 as well.

**Review by spark0510 — CHANGES_REQUESTED** [review]
Here's some comments with the procedure.

1. For whom use our default FaaSr container, they don't really need their own ECR. So I suggest to only create the repository when they provide their own image from other than ECR. (If they provide the image with their own ECR, we don't need to create one for them as they already have it)
2. I found that repository name is the image uri. e.g., if the repository name is "FaaSr", then the image url will be "<my_id>.dkr.ecr.<region>.amazonaws.com/**FaaSr**". 
So, I suggest to set the repository name with the user provided name. e.g., if user gives us "spark0510/test" for their image name from docker, we can build the repository with the name "<my_id>.dkr.ecr.<region>.amazonaws.com/**test**", instead of asking users to provide the repository name i.e., "lambda-script"
3. I suggest to just create the role name "faasr_role" and explictly use in our FaaSr Lambda functions, instead of asking users to provide the role name. i.e. faasr-lambda
4. I can see "sts()" functions many times, however, it seems like it's only for the account_id. I suggest to use only once in the "faasr_register_workflow_aws_lambda" on the corresponding lambda account. As a result, it will reduce the amount of transactions.

File: R/faasr_client_api_aws-lambda.R (hunk 4 above)
Comment (spark0510) [reply R/faasr_client_api_aws-lambda.R]: not 666.

File: R/faasr_client_api_aws-lambda.R (hunk 1 above)
Comment (nanmu08) [reply R/faasr_client_api_aws-lambda.R:55]: Has updated

File: R/faasr_client_api_aws-lambda.R (hunk 2 above)
Comment (nanmu08) [reply R/faasr_client_api_aws-lambda.R]: has updated

File: R/faasr_client_api_aws-lambda.R (hunk 3 above)
Comment (nanmu08) [reply R/faasr_client_api_aws-lambda.R]: has updated

**Comment by nanmu08** [comment]
Updated lambda register process. remove Image build, support create lambda function in multiple region.

**Comment by spark0510** [comment]
Tested with forcast example: working fine.

**Pull request merged**